packages/lab/src/components/Flow/DroppableFlow.tsx

```diff
@@ -0,0 +1,156 @@
+import { useCallback, useState } from "react";
+
+import ReactFlow, {
+  Connection,
+  EdgeChange,
+  NodeChange,
+  ReactFlowProps,
+  addEdge,
+  applyEdgeChanges,
+  applyNodeChanges,
+  useReactFlow,
+} from "reactflow";
+import "reactflow/dist/style.css";
+
+import { DragEndEvent, useDndMonitor, useDroppable } from "@dnd-kit/core";
+
+import { v4 as uuidv4 } from "uuid";
+
+import { ExtractNames, useUniqueId } from "@hitachivantara/uikit-react-core";
+
+import { HvFlowEdge, HvFlowNode } from "./types";
+import { staticClasses, useClasses } from "./Flow.styles";
+import { useFlowContext } from "./FlowContext";
+
+export { staticClasses as flowClasses };
+
+export type HvFlowClasses = ExtractNames<typeof useClasses>;
+
+export interface HvDroppableFlowProps<
+  NodeData = any,
+  NodeType extends string | undefined = string | undefined
+> extends Omit<ReactFlowProps, "nodes" | "edges" | "nodeTypes"> {
+  /** Flow content: background, controls, and minimap. */
+  children?: React.ReactNode;
+  /** Flow nodes. */
+  nodes?: HvFlowNode<NodeData, NodeType>[];
+  /** Flow edges. */
+  edges?: HvFlowEdge[];
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvFlowClasses;
+  /** Function called when the flow changes. Returns the updated nodes and edges. */
+  onFlowChange?: (
+    nodes: HvFlowNode<NodeData, NodeType>[],
+    edges: HvFlowEdge[]
+  ) => void;
+}
+
+export const HvDroppableFlow = ({
+  id,
+  className,
+  children,
+  onFlowChange,
+  classes: classesProp,
+  nodes: initialNodes = [],
+  edges: initialEdges = [],
+  onConnect: onConnectProp,
+  onNodesChange: onNodesChangeProp,
+  onEdgesChange: onEdgesChangeProp,
+  ...others
+}: HvDroppableFlowProps) => {
+  const { classes, cx } = useClasses(classesProp);
+
+  const elementId = useUniqueId(id, "hvFlow");
+
+  const reactFlowInstance = useReactFlow();
+
+  const { nodeTypes } = useFlowContext();
+
+  const [nodes, setNodes] = useState(initialNodes);
+  const [edges, setEdges] = useState(initialEdges);
+
+  const { setNodeRef } = useDroppable({
+    id: elementId,
+  });
+
+  const handleDragEnd = useCallback(
+    (event: DragEndEvent) => {
+      if (event.over && event.over.id === elementId) {
+        const type = event.active.id.toString();
+
+        // Converts the coordinates to the react flow coordinate system
+        const position = reactFlowInstance.project({
+          x: (event.active.data.current?.hvFlow?.x || 0) - event.over.rect.left,
+          y: (event.active.data.current?.hvFlow?.y || 0) - event.over.rect.top,
+        });
+
+        const newNode: HvFlowNode = {
+          id: uuidv4(),
```

Comment:
`uuid` seems overkill. it's huge, slower, and gives security guarantees we don't care about
we could use `uid`

Reply:
Great, I'll update the this library instead of uuid!